htlcswitch/circuit_map.go

```diff
@@ -225,8 +225,10 @@ func (cm *circuitMap) initBuckets() error {
 
 // restoreMemState loads the contents of the half circuit and full circuit
 // buckets from disk and reconstructs the in-memory representation of the
-// circuit map.  Afterwards, the state of the hash index is reconstructed using
-// the recovered set of full circuits.
+// circuit map. Afterwards, the state of the hash index is reconstructed using
+// the recovered set of full circuits. This method will also remove any stray
+// keystones, which are those that appear fully-opened, but have no pending
+// circuit related to the intended incoming link.
```

Comment:
Perhaps this is also an opportunity to move this from the constructor of the switch to the `Start()` method? As is, this can slow down initial start up a bit, since we block on the _creation_ of the server for all these circuits to be loaded. If a node has a set of dangling circuits, then there can potentially be thousands of them. However, it may be the case that this PR, along with its sibling PRs also help to clean up all the circuit garbage.

Reply:
All responses need to be reloaded into mailboxes before any links are added, it's one of the  constraints that guarantees correctness in reconstructing our in-mem state. We could do something that delays the adding of other links after `Start` returns, but it would be a bit more involved

Reply:
Gotcha makes sense.